### PR TITLE
[WIP] Adds autocomplete for keyboard, subproject, keymap for make

### PR DESCRIPTION
**Pull request by jackhumbert**
You can add it to your session with:

    complete -W "`make autocomplete NO_VERSION=yes`" make

I think there's a better way to do it that uses [actual bash scripts](http://fahdshariff.blogspot.com/2011/04/writing-your-own-bash-completion.html) - if anyone else wants to give this a shot, feel free!

### CODE REVIEW

**Comment by dungdung** [comment]
I'm only getting `keyboard-keymap` and no subprojects. 

`make planck-re\<tab\>` doesn't give any option

File: Makefile [hunk 1]
```diff
@@ -564,6 +566,50 @@ git-submodule:
...
+    
+    KEYMAPS_OPTION := $$(sort $$(KEYMAPS_OPTION) $$(LAYOUT_KEYMAPS_OPTION))
+
+    $$(foreach KEYMAP_OPTION, $$(KEYMAPS_OPTION), $$(eval $$(call PRINT_OPTION)))
+endef
+
+define ALL_SUBPROJECTS
+    SUBPROJECTS_OPTION := $$(notdir $$(patsubst %/rules.mk,%,$$(wildcard $(ROOT_DIR)/keyboards/$$(KEYBOARD_OPTION)/*/rules.mk)))
+    $$(eval $$(call ALL_KEYMAPS))
+    ifneq ($$(SUBPROJECT_OPTION),)
```
Comment (dungdung) [inline Makefile]: Found the issue, this line should check for `SUBPROJECTS_OPTION` instead

Comment (jackhumbert) [reply Makefile]: Ah, yeah - thanks!

**Comment by jasonm23** [comment]
Maybe add targets too? Dfu or avrdude

**Comment by jackhumbert** [comment]
We're gonna be going in a different direction with the build system soon, so this wouldn't be use!

**Pull request closed without merging**